nav2_collision_monitor/src/collision_detector_node.cpp

```diff
@@ -253,26 +258,42 @@ bool CollisionDetector::configureSources(
         rclcpp::ParameterValue("scan"));  // Laser scanner by default
       const std::string source_type = get_parameter(source_name + ".type").as_string();
 
+      nav2_util::declare_parameter_if_not_declared(
+        node, source_name + ".source_timeout",
+        rclcpp::ParameterValue(source_timeout.seconds()));    // node source_timeout by default
+      const rclcpp::Duration sensor_specific_source_timeout = rclcpp::Duration::from_seconds(
+        get_parameter(
+          source_name + ".source_timeout").as_double());
+
+      nav2_util::declare_parameter_if_not_declared(
+        node, source_name + ".block_if_invalid",
+        rclcpp::ParameterValue(block_if_invalid));    // node block_if_invalid by default
+      const bool sensor_specific_block_if_invalid =
+        get_parameter(source_name + ".block_if_invalid").as_double();
+
```

Comment:
The same here: please move this logic inside `Source` class

Reply:
Same as above. Allow both per global and per sensor config